pkg/manager/member/tikv_scaler.go

```diff
@@ -103,16 +103,30 @@ func (tsd *tikvScaler) ScaleIn(meta metav1.Object, oldSet *apps.StatefulSet, new
 		return fmt.Errorf("tikvScaler.ScaleIn: failed to get pods %s for cluster %s/%s, error: %s", podName, ns, tcName, err)
 	}
 
+	tc, _ := meta.(*v1alpha1.TidbCluster)
+	upNumber := 0
+	storeState := ""
+	for _, store := range tc.Status.TiKV.Stores {
+		if store.State == v1alpha1.TiKVStateUp {
+			upNumber++
+		}
+		if store.PodName == podName {
+			storeState = store.State
+		}
+	}
+	if upNumber < 3 {
+		return fmt.Errorf("the number of stores in Up state of TidbCluster [%s/%s] is %d, less than 3, can't scale in TiKV, podname %s ", meta.GetNamespace(), meta.GetName(), upNumber, podName)
+	} else if upNumber == 3 {
+		if storeState == v1alpha1.TiKVStateUp {
+			return fmt.Errorf("can't scale in TiKV of TidbCluster [%s/%s], cause the number of up stores is 3, and the store in Pod %s which is going to be deleted is up too", meta.GetNamespace(), meta.GetName(), podName)
```

Comment:
ditto

Reply:
BTW, did you test this case? The store cannot change to tombstone, right?

Reply:
> BTW, did you test this case? The store cannot change to tombstone, right?

@DanielZhangQD Yep. If the store state is tombstone, it will be moved to `Status.TiKV.TombstoneStores`.